qutip/superop_reps.py

```diff
@@ -328,33 +330,36 @@ def _generalized_kraus(q_oper, thresh=1e-10):
     #       diamond norm differences between two CP maps.
     if q_oper.type != "super" or q_oper.superrep != "choi":
         raise ValueError("Expected a Choi matrix, got a {} (superrep {}).".format(q_oper.type, q_oper.superrep))
-    
+
     # Remember the shape of the underlying space,
     # as we'll need this to make Kraus operators later.
-    dL, dR = map(int, map(sqrt, q_oper.shape))
+    from math import ceil
+
     # Also remember the dims breakout.
     out_dims, in_dims = q_oper.dims
     out_left, out_right = out_dims
     in_left, in_right = in_dims
 
+    dL, dR = np.prod(in_left), np.prod(out_left)
+    dL2, dR2 = np.prod(in_right), np.prod(out_right)
+
     # Find the SVD.
     U, S, V = svd(q_oper.full())
 
     # Truncate away the zero singular values, up to a threshold.
-    nonzero_idxs = S > thresh
-    dK = nonzero_idxs.sum()
-    U = U[:, nonzero_idxs]
-    S = sqrt(S[nonzero_idxs])
+
+    dK = np.count_nonzero(S > thresh)
```

Comment:
I'm just checking why this code was changed. Yes, `svd` returns the singular values in descending order, but what was wrong with previous code that didn't rely on that?

Reply:
mmm... after looking at it the changed code here gives exactly the same result. I expect that indexing in this way instead of using a mask will be faster, but I will run some tests and check it.

Reply:
Yes in fact it runs faster